reposilite-plugins/prometheus-plugin/src/main/kotlin/com/reposilite/plugin/prometheus/PrometheusPlugin.kt

```diff
@@ -32,38 +41,77 @@ class PrometheusPlugin : ReposilitePlugin() {
             ?: throw IllegalStateException("Prometheus password is not defined")
 
         val prometheusFacade = PrometheusFacade(
-            failureFacade = facade(),
+            failureFacade = facade<FailureFacade>(),
             prometheusUser = prometheusUser,
             prometheusPassword = prometheusPassword
         )
 
+        JvmMetrics.builder().register()
+        logger.info("Prometheus | JVM metrics has been initialized.")
+
+        val reposiliteMetrics = ReposiliteMetrics.builder {
+            statisticsFacade = facade<StatisticsFacade>()
+            statusFacade = facade<StatusFacade>()
+            failureFacade = facade<FailureFacade>()
+        }.register()
```

Comment:
Is there any reason why we can't just use constructor to pass these values? Using lazy props is a bit dirty and usually it's better to avoid them

Reply:
tbh the code I copied the metrics shit from was using a builder pattern so I just copied the builder pattern to match it lol

but I can kill the builder if you'd like

Reply:
So let's kill all of them with some simple extension functions :+1:

Reply:
alright, I'll do that later when I'm back at my computer